test/models/filter_test.rb

```diff
@@ -6,4 +6,19 @@ class FilterTest < ActiveSupport::TestCase
     assert Filter
   end
 
+  test "it filters developers by tags and order" do
+    tag_1, tag_2, tag_3 = create_list(:tag,3)
+    dev_1 = create(:developer, tags: [tag_1, tag_2, tag_3])
+    dev_2 = create(:developer, tags: [tag_1, tag_2])
+    dev_3 = create(:developer, tags: [tag_2, tag_3])
+
+    params = {}
+    params['Sort By'] = "Rate: Descending"
+    params['filter'] = {tag_1.name =>"0", tag_2.name =>"1", tag_3.name=>"1"}
```

Comment:
Surrounding space missing for operator `=>`.<br>Space inside { missing.<br>Space inside } missing.<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.